Team8923_CENTERSTAGE/src/main/java/org/firstinspires/ftc/team8923_CENTERSTAGE/BaseTeleOp.java

```diff
@@ -1,8 +1,68 @@
 package org.firstinspires.ftc.team8923_CENTERSTAGE;
 
-
 abstract public class BaseTeleOp extends BaseOpMode {
-    // add mechanism speeds and drive methods when mechanisms is ready
-    double slidesSpeed = 0.6;
-    double releaseSpeed = 0.9;
+
+    private boolean flipGondola = false;
+
+    private Toggle pressToggle = new Toggle();
+
+    public void driveMechanism() {
+        /* controls:
+        hold left bumper: intake runs in, pixel release rotates in
+        hold right bumper: intake runs out, pixel release rotates out
+        hold Y button: pixel release spins out
+        right trigger: intake runs outward
+        d-pad up and down: linear slides up and down correspondingly
+        B button: gondola flip out
+        A button: gondola returns to initialized position (down/in)
+        press left joystick down: release drone
+         */
+
+        // intake and servo inside gondola rotate together inside when left bumper, out when right bumper
+        if (gamepad2.left_bumper) {
+            motorIntakeWheels.setPower(0.5);
+            servoReleasePixel.setPower(1.0);
+        } else if (gamepad2.right_bumper) {
+            motorIntakeWheels.setPower(-1.0);
+            servoReleasePixel.setPower(-1.0);
+        } else {
+            motorIntakeWheels.setPower(0.0);
+            servoReleasePixel.setPower(0.0);
+        }
+
+        // button Y runs pixel release servo without intake running
+        if (gamepad2.y) {
+            servoReleasePixel.setPower(-1.0);
+        } else {
+            servoReleasePixel.setPower(0.0);
+        }
+
+        // slides controlled by up and down dpad
+        if (gamepad2.dpad_up) {
+            motorSlides.setPower(1.0);
+        } if (gamepad2.dpad_down) {
+            motorSlides.setPower(0.0);
+        } else {
+            motorSlides.setPower(0.0);
```

Comment:
Good point, it should be setting the power to -1. Also, our slides are now powered by servos instead of a motor so I actually changed this entire part to adapt to the hardware changes